File: pyg_sellergoods_interface/src/main/java/com/pyg/sellergoods/service/SpecificationService.java
package com.pyg.sellergoods.service;

import bean.PageResult;
import com.pyg.pojo.TbSpecification;
import groupEntity.Specification;

import java.util.List;
import java.util.Map;

/**
 * @author dibulidubulijiojio
 * @version v1.0
 * @date 2019/3/8  10:11
 * @description TODO
 **/
public interface SpecificationService {
    public List<TbSpecification>  findAll();

    PageResult findPage(Integer pageNo, Integer pageSize);

    void add(Specification specification);

    Specification findOne(Long id);

    void update(Specification specification);

    void deleteByIds(Long[] ids);

    PageResult findPage(Integer pageNo, Integer pageSize, TbSpecification tbSpecification);

    List<Map> findSpecList();
}

File: pyg_manager_web/src/main/webapp/js/service/specificationService.js
//只和后台交互 不和 html交互
app.service("specificationService", function ($http) {

    this.findSpecList = function () {
        return $http.get("../specification/findSpecList");
    }

    this.search = function (pageNo, pageSize, searchEntity) {
        return $http.post("../specification/search?pageNo=" + pageNo + "&pageSize=" + pageSize, searchEntity);
    }

    this.findPage = function (pageNo, pageSize) {
        return $http.get("../specification/findPage?pageNo=" + pageNo + "&pageSize=" + pageSize);
    }

    this.findAll = function () {
        return $http.get("../specification/findAll");
    }

    this.add = function (entity) {
        return $http.post("../specification/add", entity);
    }

    this.update = function (entity) {
        return $http.post("../specification/update", entity);
    }

    // 根据id查询对象
    this.findOne = function (id) {
        return $http.get("../specification/findOne?id=" + id);
    }

// 	       删除
    this.dele = function (selectIds) {
        return $http.get("../specification/dele?ids=" + selectIds);
    }

})
